### PR TITLE
Add webhook for NnfDataMovementProfile

### PR DESCRIPTION
**Pull request by bdevcich**

### CODE REVIEW

File: api/v1alpha1/nnfdatamovementprofile_webhook.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+var nnfdatamovementprofilelog = logf.Log.WithName("nnfdatamovementprofile-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *NnfDataMovementProfile) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
```
Comment (matthew-richerson) [inline api/v1alpha1/nnfdatamovementprofile_webhook.go:44]: Do you want to remove these TODOs?

File: api/v1alpha1/nnfdatamovementprofile_webhook.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	if err := r.validateContent(); err != nil {
+		nnfdatamovementprofilelog.Error(err, "invalid NnfDataMovementProfile resource")
+		return nil, err
+	}
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *NnfDataMovementProfile) ValidateDelete() (admission.Warnings, error) {
+	nnfdatamovementprofilelog.Info("validate delete", "name", r.Name)
```
Comment (matthew-richerson) [inline api/v1alpha1/nnfdatamovementprofile_webhook.go:101]: The "delete" verb isn't listed in the kubebuilder line, so this won't get run, right? Putting the log message in there is a little misleading.

File: api/v1alpha1/nnfdatamovementprofile_webhook.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var nnfdatamovementprofilelog = logf.Log.WithName("nnfdatamovementprofile-resource")
```
Comment (matthew-richerson) [inline api/v1alpha1/nnfdatamovementprofile_webhook.go:35]: I know this was copied from the other webhooks, but the "-resource" seems redundant

File: api/v1alpha1/nnfdatamovementprofile_webhook_test.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+		nnfProfile = nil
+	})
+
+	It("Should not allow modification of Data in a pinned resource", func() {
+		nnfProfile.ObjectMeta.Name = pinnedResourceName
+		nnfProfile.ObjectMeta.Namespace = otherNamespaceName
+		nnfProfile.Data.Pinned = true
+
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+		Eventually(func() error {
```
Comment (matthew-richerson) [inline api/v1alpha1/nnfdatamovementprofile_webhook_test.go:121]: Can you wrap all the Get()s like this? It's rare that they fail, but every once in a while the timing is just right where a straight Expect() will return an error.

File: api/v1alpha1/nnfdatamovementprofile_webhook_test.go [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+			return k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nnfProfile), nnfProfile)
+		}).Should(Succeed())
+
+		Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nnfProfile), newProfile)).To(Succeed())
```
Comment (matthew-richerson) [inline api/v1alpha1/nnfdatamovementprofile_webhook_test.go:125]: Why do you have to Get() it again?

**Review by matthew-richerson — APPROVED** [review]

File: api/v1alpha1/nnfdatamovementprofile_webhook.go (hunk 3 above)
Comment (bdevcich) [reply api/v1alpha1/nnfdatamovementprofile_webhook.go:35]: I'll remove it from the others as well.

File: api/v1alpha1/nnfdatamovementprofile_webhook_test.go (hunk 4 above)
Comment (bdevcich) [reply api/v1alpha1/nnfdatamovementprofile_webhook_test.go:121]: Sure - all the pinned/default stuff is ripped off from the storage profile tests.

**Review by roehrich-hpe — APPROVED** [review]

**Pull request merged**